mmtk/src/api.rs

```diff
@@ -226,6 +226,16 @@ pub extern "C" fn mmtk_post_alloc(
     bytes: usize,
     semantics: AllocationSemantics,
 ) {
+    #[cfg(all(feature = "object_pinning", not(feature = "non_moving")))]
```

Comment:
Did you include this code when you ran benchmarks in https://github.com/mmtk/mmtk-julia/pull/158#issuecomment-2199302040?

This should be fixed in mmtk-core by bulk clearing the pinning bits. We don't want to clear the bits here for every object.

Reply:
No, I didn't. Is it fixed in the mmtk-core version we're currently using? 
I'll remove that code then and maybe try to re-run the benchmarks in the meantime.

Reply:
I don't think we have fixed it in mmtk-core. But we should fix it in mmtk-core.

Reply:
You also only clear the bits in this function, not in the JIT'd code. So for objects allocated in the JIT'd code, the pinning bits are not cleared properly.

Reply:
> You also only clear the bits in this function, not in the JIT'd code. So for objects allocated in the JIT'd code, the pinning bits are not cleared properly.

I think I'm clearing them in the slowpath, not sure if they need to be cleared in the fastpath as well (since they will be bulk cleared for the memory chunk I get in the slowpath).

Reply:
There is no slowpath or fastpath for `post_alloc`. We either inline them, or not inline them. Both versions should be the same.

Reply:
What I meant was that in the allocation slowpath, you could clear the bits by calling `mmtk_post_alloc`, and it would be cleared for the chunk of memory you get between the new cursor and limit. But I checked the Julia [PR](https://github.com/mmtk/julia/pull/59/files#diff-42a3fc6ebd78315e40602ab4108f5cd380ada14ff81987b696d7b4ee9372a4de) and I wasn't calling `mmtk_post_alloc` there either, since I'm now inlining the code to set the VO bit. So, in fact, that code to bulk zero the pinning bits was not being executed in the results above.

Reply:
I see you removed the code for the pin bits. Just make sure that we are on the same page. With this PR, the pin bits are not cleared, right? It sounds fine to me. But it is an issue that we will need to fix.

Reply:
That's right. It should be fixed inside mmtk-core. I'm currently looking into it and will open a PR there instead.